src/web/js/output-ui.js

```diff
@@ -20,58 +20,153 @@
     srcloc = runtime.getField(srclocLib, "values");
     ED = runtime.getField(errordisplayLib, "values");
     PP = runtime.getField(parsePyret, "values");
-
+    
     // TODO(joe Aug 18 2014) versioning on shared modules?  Use this file's
     // version or something else?
     var shareAPI = makeShareAPI("");
+    
+    var highlightedPositions = [];
 
-    var marksByEditor = {}; // Map from editor-source to {mark-span, callback}
-    function addMark(source, editor, from, to, opts, cb) {
-      if (!marksByEditor[source])
-        marksByEditor[source] = [];
-      if (editor === undefined) {
-        console.trace();
-        console.error("Editor is undefined ", source, from, to, opts, cb);
-      }
-      marksByEditor[source].push({editor: editor, from: from, to: to, opts: opts, cb: cb});
-    }
-
-    function runMarks() {
-      var allMarks = marksByEditor;
-      var chunkSize = 100;
-      marksByEditor = {};
-      for (var source in allMarks) {
-        var marks = allMarks[source];
-        var editor = marks[0].editor; // these should be uniformly the same editor...
-        for (var i = 0; i < Math.ceil(marks.length / chunkSize); i++) {
-          function chunk(start, editor, marks) {
-            return function() {
-              editor.operation(function() {
-                for (var j = start; j < marks.length && j < start + chunkSize; j++) {
-                  //...but we'll use the right editor, just in case
-                  var mark = marks[j];
-                  var handle = mark.editor.markText(mark.from, mark.to, mark.opts);
-                  if (mark.cb)
-                    mark.cb(mark.editor, handle);
-                }
-              });
-            };
-          }
-          util.suspend(chunk(i * chunkSize, editor, marks));
+    var Position = function() {
+      
+      function cached_find(doc, positionCache, textMarker) {
+        var changeGeneration = doc.changeGeneration();
+        if (positionCache.has(changeGeneration))
+          return positionCache.get(changeGeneration);
+        else {
+          var pos = textMarker.find();
+          positionCache.set(changeGeneration, pos);
+          return pos;
         }
       }
-    }
-
-
-    function mapK(inList, f, k, outList) {
-      if (inList.length === 0) { k(outList || []); }
-      else {
-        var newInList = inList.slice(1, inList.length);
-        f(inList[0], function(v) {
-          mapK(newInList, f, k, (outList || []).concat([v]))
+    
+      function Position(doc, source, from, to, inclusiveLeft, inclusiveRight) {
+        var inclusiveLeft   = false,
+            inclusiveRight  = false;
```

Comment:
@jpolitz Good catch! `Position` used to take an option object with fields of those names, but when I added them as parameters, I did not rename the local variables. (This doesn't cause any bugs, since `Position` is always instantiated with exclusive bounds at the moment, but that will change.)